R/as_raster.R

```diff
@@ -1,26 +1,29 @@
-#' @export
-as.raster.rasterizeMatrix <- function(x, color = c('lightblue','darkblue'), span = 50,
-                                      zeroIgnored = TRUE, image = NULL, background = "#FFFFFF",
-                                      alpha = 255, layout = c("weighted", "cover")) {
+as_raster <- function(x, ...) {
+  UseMethod("as_raster", x)
+}
 
+as_raster.matrix <- function(x, color = c('lightblue','darkblue'), span = 50,
```

Comment:
For conventional naming styles of R functions, would it be better to only have underscores in function names? In this case we would rename this `as_raster_matrix` rather than `as_raster.matrix` for consistency.

Reply:
Aha, hi Hammad!
`as_raster.matrix()` is considered as a `S3` method and `matrix` here is used to define class. So does `as_raster.list()`. The benefit of doing so is that I only need to call `as_raster(obj)` and the class of obj (either `list` or `matrix`) will be despatched to corresponding functions.

Reply:
I see, that makes sense. Thanks for clearing it up 👍 .